components/CountryNavigator.js

```diff
@@ -0,0 +1,248 @@
+
+import { nth } from "../utils";
+import { RainbowLink } from "../components/RainbowLink";
+import Legend from "../components/Legend";
+import { CATEGORIES_ORDERED_LIST, RAINBOW_COLORS } from "../constants";
+import { getData } from "../data";
+import { useFlagDimensions } from "../hooks/useFlagDimensions";
+import { Flag } from "../components/Flag";
+
+const CountryNavigator = () => {
+  const [year, setYear] = useState(2024);
+  const [sortDict, setSortDict] = useState({
+    type: "ranking",
+    ascending: true,
+  });
+
+  const data = useMemo(
+    () => getData({ years: [year], sortingParams: sortDict }),
+    [year, sortDict]
+  );
+
+  const countries = data.map((d) => d.country);
+  const [currentCountryIndex, setCurrentCountryIndex] = useState(0);
+  const dataCountry = useMemo(() => getData({ countries: countries }), [
+    countries,
+  ]);
+  const [isRunning, setIsRunning] = useState(false);
+
+  useEffect(() => {
+    let interval;
+    if (isRunning) {
+      const updateCountry = () => {
+        setCurrentCountryIndex(
+          (prevIndex) => (prevIndex + 1) % countries.length
+        );
+      };
+
+      interval = setInterval(updateCountry, 1000);
+    }
+
+    return () => clearInterval(interval);
+  }, [isRunning]);
+
+  const currentCountry = countries[currentCountryIndex];
+
+  const currentCountryData = dataCountry.filter(
+    (item) => item.country === currentCountry
+  );
+
+  const years = dataCountry.map((d) => d.year).sort();
+  const uniqueYears = [...new Set(years)];
+
+  const lastYear = [uniqueYears[uniqueYears.length - 1]];
+  const allButLast = uniqueYears.slice(0, uniqueYears.length - 1);
+  const [flagsContainer, flagDimensions] = useFlagDimensions(1.3, 0);
+  const [allFlagsContainer, allFlagDimensions] = useFlagDimensions(
+    allButLast.length,
+    null
+  );
+
+  const handleStart = () => {
+    setIsRunning(true);
+  };
+
+  const handleStop = () => {
+    setIsRunning(false);
+  };
+
+  const handleNextCountry = () => {
+    setCurrentCountryIndex((prevIndex) => (prevIndex + 1) % countries.length);
+  };
+
+  const handlePreviousCountry = () => {
+    setCurrentCountryIndex((prevIndex) =>
+      prevIndex === 0 ? countries.length - 1 : prevIndex - 1
+    );
+  };
+  const SortButton = styled(Button)(({ theme }) => ({
+    borderColor: " #cbd5e1",
+    color: " #cbd5e1",
+    "&:hover": {
+      borderColor: "#f8fafc",
+      color: "#f8fafc",
+      boxShadow: "none",
+    },
+    "&:active": {
+      boxShadow: "none",
+      color: "#f8fafc",
+      borderColor: "#f8fafc",
+    },
+    "&:focus": {
+      boxShadow: "0 0 0 0.2rem rgba(0,123,255,.5)",
+    },
+  }));
+  const colorScale = scaleOrdinal({
+    domain: CATEGORIES_ORDERED_LIST,
+    range: RAINBOW_COLORS,
+  });
+
+  return (
+    <div className={`w-full mx-auto `}>
+      <Head>
+        <title>{`${
+          countries[currentCountryIndex] ?? "Loading"
```

Comment:
I guess that we can replace all these `countries[currentCountryIndex]` with the `currentCountry` that was defined earlier to make it simpler :)